packages/plugin-core/src/_extension.ts

```diff
@@ -848,6 +860,44 @@ export async function showLapsedUserMessage(assetUri: vscode.Uri) {
     });
 }
 
+export function shouldDisplayMissingDefaultConfigMessage(opts: {
+  ext: IDendronExtension;
+  extensionInstallStatus: InstallStatus;
+}): boolean {
+  if (opts.extensionInstallStatus === InstallStatus.UPGRADED) {
+    const wsRoot = opts.ext.getDWorkspace().wsRoot;
+    const rawConfig = DConfig.getRaw(wsRoot);
+    const resp = ConfigUtils.detectMissingDefaults({ config: rawConfig });
+    return resp.data !== undefined && resp.data.needsBackfill;
+  } else {
+    return false;
+  }
+}
+
+export function showMissingDefaultConfigMessage(opts: {
+  ext: IDendronExtension;
+}) {
+  AnalyticsUtils.track(ConfigEvents.ShowMissingDefaultConfigMessage);
+  const ADD_CONFIG = "Add Missing Configuration";
+  vscode.window
+    .showInformationMessage(
+      "We have detected a missing default configuration. Would you like to add them to dendron.yml?",
```

Comment:
This could also be prompted if the user accidentally removes a config that should exist (either because it's not optional, or it's optional but has a default value). In that case, the suggestion would be a bit misleading.

How about `We have detected a missing configuration. This may be caused by a newly introduced configuration, or by deleting an existing configuration that has a default value. Would you like to add them to dendron.yml?`

It's a mouthful, but less confusing.